### DISCUSSION TITLE
Default financial report - Issue with bills paid after the default date

### DISCUSSION THREAD

**Discussion by top-cg** [firefly-iii/firefly-iii#6095] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I discovered the "default financial report" has a strange behavior when it comes to "bills" : 
- in the default financial report, the bills appears to be "not paid" if the transaction was recorded too far from the expected date.
- when in the bill page ... it was properly identified as "paid"

Here is a good example in April 2022.

This is a mobile phone bill, very classic, monthly.
Configured with a "default" date : 28th every month.

Properly identified as paid.
![image](https://user-images.githubusercontent.com/18399003/169748982-9f77891e-3dda-408b-960a-e80696a97bc5.png)

Yet in the default financial report, this isn't ...
![image](https://user-images.githubusercontent.com/18399003/169749172-37d693cf-b918-4d24-b091-a55b19e0d872.png)

Let's change the default date to 25 ?
![image](https://user-images.githubusercontent.com/18399003/169749187-f08d543b-50e2-4a6e-aee8-db326b9bbc86.png)

Let's refresh the report ... tadaaaa ... paid
![image](https://user-images.githubusercontent.com/18399003/169749217-6722e734-a7d6-4c10-8199-38f855b7229e.png)

So i played ...
- the invoice was truly paid the 25th April 2022
- default bill date "25" => works
- default bill date "26" => fail
- default bill date "27" => fail
- default bill date "24" => works
- default bill date "4" => works

In this case, this is clear for me : the financial report will consider the bill "not paid" if the transaction happen "BEFORE the default billing date".

I have such issue on several bills ... always the same : the transaction happen after the "defaut billing date".

### Expected behaviour

If the "Bills" page is capable to properly identify a transaction was paid on the month whatever is the date ... the default financial report should also.

Maybe the default report shall not consider the "default" billing date (should use the last day of the month instead) has the date the bill was paid has no importance (paid / not paid is the key factor).

### Debug information

Debug information generated at 2022-05-23 07:37:52 Europe/Paris for Firefly III version **5.7.6**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.6 |
| Firefly III API | 1.5.6 |
| Build | 185, 18-05-2022 06:54:00 CEST |
| DB version | 18 (exp. 18) |
| PHP | `8.0.17` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Paris |
| Browser TZ | Europe/Paris |
| App environment | local |
| App debug mode | true |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | fr_FR |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-05-01 00:00:00 |
| Session end | 2022-05-31 23:59:59 |
| Session first | 2020-12-11 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:100.0) Gecko/20100101 Firefox/100.0 |

### Steps to reproduce

See description.

**Comment by top-cg** [comment]
Ah i forget ... up until the app is corrected (if you judged it needed), a simple "user" quick fix is just changing the default date of the bill.
This is working well if bills are not changing drastically of date each month (unfortunately i have such case ... so i set them on 27th to avoid February being an issue).

**Comment by JC5** [comment]
> In this case, this is clear for me : the financial report will consider the bill "not paid" if the transaction happen "BEFORE the default billing date".

This is indeed how it works.

**Comment by top-cg** [comment, marked as answer]
Hello,

Thanks for the answer.

May I ask you why you find it more logical like this as I understand this is intended.

I found it misleading ... For example I paid our nanny in advance this month by 2 weeks ... As a result this is considered as not paid in the report but it is. And if I'm changing the default date of the bill to 15th ... This is defeating the main goal of these dates which is "giving an idea of the best landing date to ensure the current account is enough full to pay coming bills".

As there is a difference in the approach in the report and in the rest of the app on that point... You certainly use the report in a way I don't see ;)

**Reply by JC5** [reply]
I see your point. I'll see what I can do about the report. Since you pay the nanny "this month" it's easiest to set the date of the bill to the first of the month.

**Reply by top-cg** [reply]
Not a big deal ... I can live with it when sometimes 😊
If I found the time, I will suggest an update to be added in the documentation to mention the logic of the report and the workaround.
